src/value.rs

```diff
@@ -190,11 +218,12 @@ impl ValueCommitment {
         let hasher = pallas::Point::hash_to_curve("z.cash:Orchard-cv");
         let V = hasher(b"v");
         let R = hasher(b"r");
+        let value = i64::try_from(value.0).expect("value must be in valid range");
```

Comment:
This restricts `ValueSum`s to an `i64` range, which is smaller than the `ValueSum` range, and thus can panic here. Instead, use `value.0.is_negative()` below as before, and here use:
```suggestion
        let abs_value = u64::try_from(value.0.abs()).expect("value must be in valid range");
```
This will always work, as `u64` is larger than the range of `|ValueSum|`.